src/DotLiquid/Util/StrFTime.cs

```diff
@@ -18,85 +20,197 @@ public static class StrFTime
+            }
+            else if ("N".Equals(directive) && (width == 3 || width == 6))
+            {
+                result = "" + width + directive;
+            }
+            return result;
+        }
 
-            for (int n = 0; n < pattern.Length; n++)
+        private static String ApplyFlag(String flag, int padwidth, String str)
+        {
+            var result = str;
+            switch (flag)
             {
-                string s = pattern.Substring(n, 1);
+                case "-": //don't pad a numerical output
+                    result = str.TrimStart('0');
+                    break;
+                case "_": //use spaces for padding
+                    result = str.TrimStart('0').PadLeft(padwidth, ' ');
+                    break;
+                case "0": //use zeros for padding
+                    result = str.TrimStart('0').PadLeft(padwidth, '0');
+                    break;
+                case "^": //upcase the result string
+                    result = str.ToUpper();
+                    break;
+                case ":": // handled by PreProcessDirective
+                    break;
+                case "#": // not implemented
+                    break;
+                    // default: do nothing.
+            }
+            return result;
+        }
 
-                if (s == "%" && pattern.Length > ++n)
-                    if (OffsetFormats.ContainsKey(pattern.Substring(n, 1)))
-                        output.Append(OffsetFormats[pattern.Substring(n, 1)].Invoke(dateTime));
-                    else if (Formats.ContainsKey(pattern.Substring(n, 1)))
-                        output.Append(Formats[pattern.Substring(n, 1)].Invoke(dateTime.DateTime));
-                    else
-                        output.Append("%" + pattern.Substring(n, 1));
-                else
-                    output.Append(s);
+        /// <summary>
+        /// ISO 8601 week-based year and week number:
+        /// - The week 1 of YYYY starts with a Monday and includes YYYY-01-04.
+        /// - The days in the year before the first week are in the last week of the previous year.
+        /// For example 2012-12-31 is in the first week of 2013 (according to ISO-8601)
+        /// <param name="dateTime"></param>
+        /// <param name="directive"></param>
+        /// </summary>
+        private static string GetIso8601WeekOfYear(this DateTime dateTime, string directive)
+        {
+            // If its Monday, Tuesday or Wednesday, then it'll be the same week
+            // as whatever Thursday, Friday or Saturday are.
+            DayOfWeek day = CultureInfo.InvariantCulture.Calendar.GetDayOfWeek(dateTime);
+            if (day >= DayOfWeek.Monday && day <= DayOfWeek.Wednesday)
+            {
+                dateTime = dateTime.AddDays(3);
             }
 
-            return output.ToString();
+            switch (directive)
+            {
+                case "G":
+                    return dateTime.ToString("yyyy");

```

Comment:
likely should have ", CultureInfo.CurrentCulture" or some explicit culture format specification.

Reply:
All occurrences of DateTime.ToString() updated to include CultureInfo.CurrentCulture.